pkg/middleware/headers.go

```diff
@@ -39,6 +40,14 @@ func newStripHeaders(headers []options.Header) alice.Constructor {
 	}
 }
 
+func flattenHeaders(headers http.Header) {
+	for name, values := range headers {
```

Comment:
I think this is fine, right? I just always get uncomfortable mutating a map inside of a loop on itself. Might be lingering paranoia from my python days where that would result in a `RuntimeError`.

The only area I had concern was the underlying `Set` might mutate the `name` with its call to `textproto.CanonicalMIMEHeaderKey` in `textproto.MIMEHeader(h)` wrapped call.

Safest route is an initial `range` on the `http.Header` to get the headers & values to mutate. Then iterating over that list and modifying the passed in `http.Header`

Reply:
This should be safe in golang. The CanonicalMIMEHeaderKey was called for both add and set. So when we hit the code here, all names should be canonicalized. I updated condition to len(values) > 1 so we only need to update when necessary and it is safe because len(nil) is  0